lightning-graph-sync/src/processing.rs

```diff
@@ -0,0 +1,591 @@
+	let has_non_incremental_updates = (non_incremental_update_presence_flag == 1);
+	// without non-incremental updates, the following fields cannot even be calculated by the server
+	let default_cltv_expiry_delta: u16 = if has_non_incremental_updates {
+		Readable::read(&mut read_cursor)?
+	} else {
+		0
+	};
+	let default_htlc_minimum_msat: u64 = if has_non_incremental_updates {
+		Readable::read(&mut read_cursor)?
+	} else {
+		0
+	};
+	let default_fee_base_msat: u32 = if has_non_incremental_updates {
+		Readable::read(&mut read_cursor)?
+	} else {
+		0
+	};
+	let default_fee_proportional_millionths: u32 = if has_non_incremental_updates {
+		Readable::read(&mut read_cursor)?
+	} else {
+		0
+	};
+	let tentative_default_htlc_maximum_msat: u64 = if has_non_incremental_updates {
+		Readable::read(&mut read_cursor)?
+	} else {
+		0
+	};
+	let default_htlc_maximum_msat = if tentative_default_htlc_maximum_msat == u64::MAX {
+		OptionalField::Absent
+	} else {
+		OptionalField::Present(tentative_default_htlc_maximum_msat)
+	};
+
+	for _current_update_index in 0..update_count {
+		// println!("update index: {}", _current_update_index);
+
+		let short_channel_id = if let Some(previous_scid) = previous_scid {
+			let scid_delta: BigSize = Readable::read(read_cursor)?;
+			previous_scid + scid_delta.0
+		} else {
+			Readable::read(read_cursor)?
+		};
+		previous_scid = Some(short_channel_id);
+
+		// let short_channel_id: u64 = Readable::read(read_cursor)?;
+		let channel_flags: u8 = Readable::read(read_cursor)?;
+
+		// flags are always sent in full, and hence always need updating
+		let standard_channel_flags = channel_flags & 0b_0000_0011;
+
+		let mut synthetic_update = if channel_flags & 0b_1000_0000 == 0 {
+			// full update
+			// println!("channel flags: {:b}, scid: {}, full update", channel_flags, short_channel_id);
+			UnsignedChannelUpdate {
+				chain_hash,
+				short_channel_id,
+				timestamp: emulated_last_seen_timestamp,
+				flags: standard_channel_flags,
+				cltv_expiry_delta: default_cltv_expiry_delta,
+				htlc_minimum_msat: default_htlc_minimum_msat,
+				htlc_maximum_msat: default_htlc_maximum_msat.clone(),
+				fee_base_msat: default_fee_base_msat,
+				fee_proportional_millionths: default_fee_proportional_millionths,
+				excess_data: vec![],
+			}
+		} else {
+			// incremental update
+			// println!("channel flags: {:b}, incremental update", channel_flags);
+
+			let read_only_network_graph = network_graph.read_only();
+			let channel = read_only_network_graph.channels().get(&short_channel_id);
+			let channel = if let Some(channel) = channel {
+				channel
+			} else {
+				return Err(LightningError {
+					err: "Couldn't find channel for update".to_owned(),
+					action: ErrorAction::IgnoreError,
+				}
+					.into());
+			};
+
+			let direction = channel_flags & 1u8;
+			let directional_info = if direction == 0 {
+				channel.one_to_two.as_ref()
+			} else {
+				channel.two_to_one.as_ref()
+			};
```

Comment:
Consider making an inline method on `ChannelInfo` for this so you can easily chain it with the previous and/or following logic.